src/sentry/snuba/metrics/fields/base.py

```diff
@@ -457,12 +487,22 @@ def generate_bottom_up_derived_metrics_dependencies(
     ) -> Iterable[Tuple[MetricOperationType, str]]:
         return [(self.metric_operation.op, self.metric_object.metric_mri)]
 
-    def build_conditional_aggregate_for_metric(self, org_id: int, entity: MetricEntity) -> Function:
+    def build_conditional_aggregate_for_metric(
+        self, org_id: int, entity: MetricEntity, query_definition: QueryDefinition
+    ) -> Function:
         snuba_function = OP_TO_SNUBA_FUNCTION[entity][self.metric_operation.op]
+        rv = self.metric_object.generate_filter_snql_conditions(org_id=org_id)
+
+        operation_based_filter = self.metric_operation.generate_filter_snql_conditions(
+            org_id=org_id, query_definition=query_definition
+        )
+        if operation_based_filter is not None:
+            rv = Function("and", [rv, operation_based_filter])
+
         return Function(
             snuba_function,
-            [Column("value"), self.metric_object.generate_filter_snql_conditions(org_id=org_id)],
-            alias=f"{self.metric_operation.op}({self.metric_object.metric_mri})",
+            [Column("value"), rv],
+            f"{self.metric_operation.op}({self.metric_object.metric_mri})",
```

Comment:
Why did you remove `alias=`?

Reply:
no reason, i moved code around and reverted things manually again